source/Weapon.cpp

```diff
@@ -101,7 +101,7 @@ void Weapon::LoadWeapon(const DataNode &node)
 		else if(key == "submunition")
 		{
 			submunitions.emplace_back(
-				GameData::Outfits().Get(child.Token(1)),
+				&GameData::Outfits().Get(child.Token(1))->GetWeapon(),
```

Comment:
IMO this is not desirable. Previously, we were storing a pointer to the key of a collection, and the choice of collection meant we had particular guarantees about that pointer (always stable, etc) regardless of the collection size. Now we're storing a pointer to an implementation detail (some field of said managed object). Probably it's fine. But it's definitely something to be aware of. 
https://github.com/endless-sky/endless-sky/blob/71b66b9c674177468d18a9289cd456d15b20dbbf/source/Weapon.h#L47-L50

I recall there were previous thoughts of registering / standardizing Weapon (i.e. `GameData::Weapons()`), but I don't have specifics on why we deferred. Likely it was related to the ability of Person ship variants to have their own specialized weapon definitions.

Reply:
Would it be better to store a pointer to the outfit?